src/vlan.rs

```diff
@@ -0,0 +1,40 @@
+use core::mem;
+
+/// VLAN tag header structure
+#[repr(C, packed)]
+#[derive(Debug, Copy, Clone)]
+#[cfg_attr(feature = "serde", derive(::serde::Serialize, ::serde::Deserialize))]
+pub struct VlanHdr {
+    /// First 2 bytes containing PCP (3 bits), DEI (1 bit), and VLAN ID (12 bits)
+    pub tci: u16,
+    /// EtherType field indicating the protocol encapsulated in the payload
+    pub eth_type: u16,
```

Comment:
You can use the `EtherType` enum directly, we do that in the `EthHdr` struct already.

```suggestion
    pub eth_type: EtherType,
```